src/client/components/dimension-measure-panel/dimension-measure-panel.tsx

```diff
@@ -39,45 +40,93 @@ export interface DimensionMeasurePanelProps {
 }
 
 export interface DimensionMeasurePanelState {
+  dividerPosition: number;
+  containerHeight: number;
+}
+
+function dividerConstraints(height: number) {
+  const minDividerPosition = Math.min(MIN_PANEL_SIZE, height);
+  const maxDividerPosition = Math.max(height - MIN_PANEL_SIZE, 0);
+  return { minDividerPosition, maxDividerPosition };
+}
+
+export function initialPosition(height: number, dataCube: DataCube) {
+  const dimensionsCount = dataCube.dimensions.size();
+  const measuresCount = dataCube.measures.size();
+  const ratio = dimensionsCount / (measuresCount + dimensionsCount);
+
+  const { minDividerPosition, maxDividerPosition } = dividerConstraints(height);
+  return clamp(height * ratio, minDividerPosition, maxDividerPosition);
 }
 
 export class DimensionMeasurePanel extends React.Component<DimensionMeasurePanelProps, DimensionMeasurePanelState> {
 
+  state: DimensionMeasurePanelState = {
+    containerHeight: 2 * MIN_PANEL_SIZE,
+    dividerPosition: MIN_PANEL_SIZE
+  };
+
+  containerRef: Element = null;
+
+  getInitialState = (container: Element) => {
```

Comment:
I think this method name is misleading as it was being used in `React.createClass` to actually return inital state.